src/pip/_internal/utils/compat.py

```diff
@@ -257,12 +257,13 @@ def ioctl_GWINSZ(fd):
             return cr
         cr = ioctl_GWINSZ(0) or ioctl_GWINSZ(1) or ioctl_GWINSZ(2)
         if not cr:
-            try:
-                fd = os.open(os.ctermid(), os.O_RDONLY)
-                cr = ioctl_GWINSZ(fd)
-                os.close(fd)
-            except Exception:
-                pass
+            if sys.platform != "win32":
```

Comment:
I think the nested if blocks can be merged here. It makes sense to semantically as well IMO (since the next block is `if not cr`)

Reply:
Updated [here](https://github.com/pypa/pip/pull/7520/files#diff-8693233a2749f69ff3bc5ba34df56e1dR259).

Reply:
Windows mypy does not like it, so backing out the change.